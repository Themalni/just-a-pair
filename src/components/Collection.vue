<template>
  <div>
    <div class="collection" id="product-grid">
      <div class="container">
        <h1 class="collection-header">{{ header }}</h1>
        <div class="row collection-items">
          <slot></slot>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: "Collection",
  props: {
    header: {
      type: String
    }
  }
};
</script>

<style lang="scss" scoped>
.collection {
  text-align: center;
  padding-top: 4em;
}
.collection-header {
  color: #333;
  font-family: $fancy-font;
  font-weight: bold;
}
.collection-items {
  margin-top: 2em;
}
</style>