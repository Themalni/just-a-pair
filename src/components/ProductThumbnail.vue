<template>
    <div class="col-md-4 col-sm-6">
        <router-link :to="'/product-view/' + id" class="product-link">
            <div class="product-thumbnail pb-2" @click="showProduct(id)">
                <div v-if="!onSale" v-show="offer" class="offer-tag" :class="{accent: onSale}">{{ offer }}</div>
                <div v-else v-show="offer" class="offer-tag" :class="{accent: onSale}">{{ salePercentage + "%" }}</div>
                <div class="product-image">
                    <img :src="image">
                </div>
                <h3 class="fancy-font mt-2 mb-2">{{ title }}</h3>
                <span class="product-price fancy-font" :class="{ 'sale-price': onSale }">${{ price }}</span>
                <span v-if="onSale" class="product-price fancy-font deep-orange-text">${{ price | salePrice(salePercentage) }}</span>
            </div>
        </router-link>
    </div>
</template>

<script>

export default {
    name: 'product-thumbnail',
    props: [
        'id',
        'image', 
        'offer', 
        'title', 
        'price', 
        'onSale', 
        'salePercentage', 
        'stock', 
        'sliderImages'
    ],
    computed: {
        products() { return this.$store.state.products; }
    },
    methods: {
        showProduct(id) {
            this.$store.dispatch('showProduct', id);
        }
    } 
}
</script>

<style lang="scss" scoped>
@import "../styles/scss/styles.scss";

.product-thumbnail {
    width: 100%;
    margin-bottom: 1.5em;
    background-color: $light-grey;
    position: relative;
    color: $dark-grey;

    &:hover {

        .product-image {
            img {
                transform: scale(1.1);
                transition: all 0.5s linear;
            }
        }
    }
}
.offer-tag {
    width: 40px;
    height: 40px;
    background-color: $main-blue;
    color: #fff;
    line-height: 2.8;
    position: absolute;
    top: 0;
    left: 0;
    z-index: 1;
}
.product-image {
    width: 100%;
    background-color: $light-grey;
    overflow: hidden;
    
    img {
        transition: all 0.5s linear;
    }
}
.product-price {
    color: grey;
    font-size: 2em;
}
.product-link {
    &:hover {
        text-decoration: none;
    }
}
.accent {
    background-color: $orange-accent;
}
.btn {
    margin: 0.5em auto;
    display: block;
}
.sale-price {
    text-decoration: line-through;
    margin: 0 0.5em;
}
</style>