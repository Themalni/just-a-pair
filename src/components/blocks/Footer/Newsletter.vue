<template>
    <div class="newsletter">
        <div class="container">
            <div class="row">
                <div class="col-md-9">
                    <p class="newsletter-text">Subscribe to our newsletter and receive a coupon for 15% off your first purchase</p>
                </div>
                <div class="col-md-3 text-right">
                    <button class="btn btn-outline-light newsletter-button">Subscribe</button>
                </div> 
            </div>
        </div>
    </div>
</template>

<script>

</script>

<style lang="scss" scoped>
@import "../../../styles/scss/styles.scss";

.newsletter {
    border-bottom: 1px solid $grey;
    padding: 30px 0;
    background-color: $main-blue;
    color: #fff;
}
.newsletter-button {
    min-width: 250px;
    font-size: 1.2em;
    box-shadow: none;
    color: #fff !important;
    border-color: #fff !important;

    &:hover {
        box-shadow: 0 1px 3px #333;
    }
}
p {
    font-size: 1.7em;
    line-height: 2;
    color: $dark-grey;
    margin-bottom: 0;
    color: #fff;
}
</style>