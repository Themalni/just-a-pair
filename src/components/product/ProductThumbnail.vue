<template>
  <div class="col-md-4 col-sm-6">
    <router-link :to="'/product/' + product.id" class="product-link">
      <div class="product-thumbnail pb-2" @click="showProduct()">
        <div v-if="!product.sale" class="offer-tag">{{ product.offer }}</div>
        <div
          v-else
          class="offer-tag"
          :class="{accent: product.sale}"
        >{{ `-${product.salePercentage}%` }}</div>
        <div class="product-image">
          <img :src="product.image">
        </div>
        <h3 class="fancy-font mt-2 mb-2">{{ product.title }}</h3>
        <span
          class="product-price fancy-font"
          :class="{ 'sale-price': product.sale }"
        >${{ product.price }}</span>
        <span
          v-if="product.sale"
          class="product-price fancy-font deep-orange-text"
        >${{ product.price | salePrice(product.salePercentage) }}</span>
      </div>
    </router-link>
  </div>
</template>

<script>
import { mapGetters, mapMutations, mapActions } from "vuex";

export default {
  name: "ProductThumbnail",
  props: {
    product: {
      type: Object
    }
  },
  computed: {
    ...mapGetters(["productPreview"])
  },
  methods: {
    ...mapActions(["clearProductPreview"]),
    showProduct() {
      this.clearProductPreview();
      this.$store.dispatch("showProduct", this.product);
    }
  }
};
</script>

<style lang="scss" scoped>
.product-thumbnail {
  width: 100%;
  margin-bottom: 1.5em;
  background-color: $light-grey;
  position: relative;
  color: $dark-grey;

  &:hover {
    .product-image {
      img {
        transform: scale(1.1);
        transition: all 0.5s linear;
      }
    }
  }
}
.offer-tag {
  width: 40px;
  height: 40px;
  background-color: $main-blue;
  color: #fff;
  line-height: 2.8;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
}
.product-image {
  width: 100%;
  background-color: $light-grey;
  overflow: hidden;

  img {
    transition: all 0.5s linear;
  }
}
.product-price {
  color: grey;
  font-size: 2em;
}
.product-link {
  &:hover {
    text-decoration: none;
  }
}
.accent {
  background-color: $orange-accent;
}
.btn {
  margin: 0.5em auto;
  display: block;
}
.sale-price {
  text-decoration: line-through;
  margin: 0 0.5em;
}
</style>
