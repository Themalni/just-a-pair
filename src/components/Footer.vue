<template>
  <div>
    <footer class="footer">
      <newsletter/>
      <div class="container links-container">
        <div class="row">
          <div class="col-md-8 footer-links">
            <div class="row">
              <div class="col-md-3 col-sm-6">
                <h4 class="links-header">Departments</h4>
                <div class="col-md-12 links">
                  <a class href="#">Women</a>
                  <a class href="#">Men</a>
                  <a class href="#">Children</a>
                  <a class href="#">Home</a>
                </div>
              </div>
              <div class="col-md-3 col-sm-6">
                <h4 class="links-header">User Profile</h4>
                <div class="col-md-12 links">
                  <a class href="#">Orders</a>
                  <a class href="#">Reviews</a>
                  <a class href="#">Coupons</a>
                </div>
              </div>
              <div class="col-md-3 col-sm-6">
                <h4 class="links-header">Help</h4>
                <div class="col-md-12 links">
                  <a class href="#">How to order</a>
                  <a class href="#">How to pay</a>
                  <a class href="#">Delivery</a>
                  <a class href="#">Returns</a>
                </div>
              </div>
              <div class="col-md-3 col-sm-6">
                <h4 class="links-header">About</h4>
                <div class="col-md-12 links">
                  <a class href="#">About Us</a>
                  <a class href="#">Privacy Policy</a>
                  <a class href="#">Career</a>
                  <a class href="#">Contact Us</a>
                </div>
              </div>
            </div>
          </div>
          <div class="col-md-4 footer-social">
            <div class="text-right">
              <a class="social-icon" href="#">
                <i class="fa fa-facebook fa-2x"></i>
              </a>
              <a class="social-icon" href="#">
                <i class="fa fa-instagram fa-2x"></i>
              </a>
              <a class="social-icon" href="#">
                <i class="fa fa-pinterest-p fa-2x"></i>
              </a>
              <a class="social-icon" href="#">
                <i class="fa fa-youtube-square fa-2x"></i>
              </a>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</template>

<script>
import Newsletter from "./Newsletter";

export default {
  name: "Footer",
  components: {
    Newsletter
  }
};
</script>

<style lang="scss" scoped>
.footer {
  background-color: $light-grey;
}
.links-container {
  padding-top: 30px;
  padding-bottom: 30px;
}
.links-header {
  font-family: $fancy-font;
  font-weight: 700;
  color: $dark-grey;
}
.links {
  padding: 0;

  a {
    display: block;
    color: $dark-grey;
    margin: 1em 0;
    font-size: 1.2em;
  }
}
.social-icon {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  background-color: $grey;
  display: inline-block;
  margin-left: 1.5em;
  padding: 7px;
  text-align: center;

  .fa {
    color: #fff;
  }

  &:hover {
    background-color: $dark-grey;
  }
}
</style>