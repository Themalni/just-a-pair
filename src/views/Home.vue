<template>
  <div class="mt-5">
    <image-grid />
    <collection header="New Collection">
      <product-thumbnail v-for="item in showThreeProducts" :key="item.id" :product="item"></product-thumbnail>
    </collection>
    <collection header="Sale">
      <product-thumbnail v-for="item in saleCollection" :key="item.id" :product="item"></product-thumbnail>
    </collection>
    <app-footer />
  </div>
</template>

<script>
import ImageGrid from "@/components/ImageGrid";
import Collection from "@/components/Collection";
import ProductThumbnail from "../components/product/ProductThumbnail";
import Footer from "@/components/Footer";

import { mapGetters } from "vuex";

export default {
  name: "Home",
  components: {
    ImageGrid,
    Collection,
    ProductThumbnail,
    "app-footer": Footer
  },
  computed: {
    ...mapGetters(["products"]),
    productList() {
      return this.products;
    },
    newCollection: function() {
      return this.productList.filter(product => {
        return product.offer === "New";
      });
    },
    showThreeProducts: function() {
      return this.newCollection.slice(0, 3);
    },
    saleCollection: function() {
      return this.productList.filter(product => {
        return product.offer === "Sale";
      });
    }
  }
};
</script>
