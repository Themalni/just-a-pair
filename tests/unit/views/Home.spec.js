import { mount, createLocalVue } from '@vue/test-utils';
import Home from '@/views/Home';
import PageHeader from "@/components/PageHeader";
import Collection from "@/components/Collection";
import ProductThumbnail from "@/components/product/ProductThumbnail";
import Footer from "@/components/Footer";
import VueScrollTo from "vue-scrollto";
import salePrice from "@/filters.js";

const localVue = createLocalVue();
localVue.directive('scroll-to', VueScrollTo);
localVue.filter('salePrice', salePrice);


describe('Home', () => {
    let wrapper

    beforeEach(() => {
        wrapper = mount(Home, {
            stubs: ['router-link'],
            localVue
        });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it('renders correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('it has PageHeader rendered in the view', () => {
        const header = wrapper.findComponent(PageHeader)
        expect(header.exists()).toBe(true)
    });

    it('it has Collection rendered in the view', () => {
        const collection = wrapper.findComponent(Collection)
        expect(collection.exists()).toBe(true)
    });

    it('it has ProductThumbnail rendered in the view', () => {
        const thumbnail = wrapper.findComponent(ProductThumbnail)
        expect(thumbnail.exists()).toBe(true)
    });

    it('it has Footer rendered in the view', () => {
        const footer = wrapper.findComponent(Footer)
        expect(footer.exists()).toBe(true)
    });
})