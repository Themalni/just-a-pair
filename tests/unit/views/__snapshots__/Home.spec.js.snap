// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div class="mt-5">
  <div class="container-fluid pr-0 pl-0 mb-4">
    <div class="row no-gutters">
      <div class="col-md-12 grid-image-container">
        <div class="grid-image-text">
          <h2 class="sub-header">Welcome to</h2>
          <h1 class="header text-uppercase font-weight-bold">Just a Pair</h1>
          <h3 class="sub-header">Your favourite shoe store online</h3> <button class="btn btn-outline-light btn-lg mt-5">New collection</button>
        </div>
      </div>
    </div>
    <div class="row no-gutters">
      <div class="col-md-4 col-sm-4"><img src="../assets/women_shoes.jpg" class="grid-image">
        <div class="grid-image-text-bottom pb-4">
          <h2 class="sub-header font-weight-bold">Women Collection</h2> <button class="btn btn-sm mt-2 button-accent">Buy now</button>
        </div>
      </div>
      <div class="col-md-4 col-sm-4"><img src="../assets/men_shoes.jpg" class="grid-image">
        <div class="grid-image-text-bottom pb-4">
          <h2 class="sub-header font-weight-bold">Men Collection</h2> <button class="btn btn-sm mt-2 button-accent">Buy now</button>
        </div>
      </div>
      <div class="col-md-4 col-sm-4"><img src="../assets/children_shoes.jpg" class="grid-image">
        <div class="grid-image-text-bottom pb-4">
          <h2 class="sub-header font-weight-bold">Children Collection</h2> <button class="btn btn-sm mt-2 button-accent">Buy now</button>
        </div>
      </div>
    </div>
  </div>
  <div class="pt-4">
    <div id="product-grid" class="collection">
      <div class="container">
        <h1 class="collection-header">New Collection</h1>
        <div class="row collection-items">
          <div class="col-md-4 col-sm-6">
            <router-link-stub to="/product/622" class="product-link">
              <div class="product-thumbnail pb-2">
                <div class="offer-tag">New</div>
                <div class="product-image"><img src="../assets/shoes_white_1.jpg"></div>
                <h3 class="fancy-font mt-2 mb-2">Star Gazers</h3> <span class="product-price fancy-font">
        $139
      </span>
                <!---->
              </div>
            </router-link-stub>
          </div>
          <div class="col-md-4 col-sm-6">
            <router-link-stub to="/product/624" class="product-link">
              <div class="product-thumbnail pb-2">
                <div class="offer-tag">New</div>
                <div class="product-image"><img src="../assets/shoes_blue_1.jpg"></div>
                <h3 class="fancy-font mt-2 mb-2">Be Diva Stillettos</h3> <span class="product-price fancy-font">
        $84
      </span>
                <!---->
              </div>
            </router-link-stub>
          </div>
          <div class="col-md-4 col-sm-6">
            <router-link-stub to="/product/626" class="product-link">
              <div class="product-thumbnail pb-2">
                <div class="offer-tag">New</div>
                <div class="product-image"><img src="../assets/shoes_grey_1.jpg"></div>
                <h3 class="fancy-font mt-2 mb-2">Symphony Light</h3> <span class="product-price fancy-font">
        $129
      </span>
                <!---->
              </div>
            </router-link-stub>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="mt-4 mb-4">
    <div id="product-grid" class="collection">
      <div class="container">
        <h1 class="collection-header">Sale</h1>
        <div class="row collection-items">
          <div class="col-md-4 col-sm-6">
            <router-link-stub to="/product/621" class="product-link">
              <div class="product-thumbnail pb-2">
                <div class="offer-tag accent">-65%</div>
                <div class="product-image"><img src="../assets/sneakers.jpg"></div>
                <h3 class="fancy-font mt-2 mb-2">Be Yourself</h3> <span class="product-price fancy-font sale-price">
        $89
      </span> <span class="product-price fancy-font deep-orange-text">
        $32
      </span>
              </div>
            </router-link-stub>
          </div>
          <div class="col-md-4 col-sm-6">
            <router-link-stub to="/product/623" class="product-link">
              <div class="product-thumbnail pb-2">
                <div class="offer-tag accent">-15%</div>
                <div class="product-image"><img src="../assets/shoes_red_1.jpg"></div>
                <h3 class="fancy-font mt-2 mb-2">Millenium Night</h3> <span class="product-price fancy-font sale-price">
        $99
      </span> <span class="product-price fancy-font deep-orange-text">
        $85
      </span>
              </div>
            </router-link-stub>
          </div>
          <div class="col-md-4 col-sm-6">
            <router-link-stub to="/product/625" class="product-link">
              <div class="product-thumbnail pb-2">
                <div class="offer-tag accent">-70%</div>
                <div class="product-image"><img src="../assets/shoes_black_1.jpg"></div>
                <h3 class="fancy-font mt-2 mb-2">Solarine</h3> <span class="product-price fancy-font sale-price">
        $249
      </span> <span class="product-price fancy-font deep-orange-text">
        $75
      </span>
              </div>
            </router-link-stub>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <footer class="footer">
      <div class="newsletter">
        <div class="container">
          <div class="row">
            <div class="col-md-9">
              <p class="newsletter-text">Subscribe to our newsletter and receive a coupon for 15% off your first purchase</p>
            </div>
            <div class="col-md-3 text-right"><button class="btn btn-outline-light newsletter-button mt-3">Subscribe</button></div>
          </div>
        </div>
      </div>
      <div class="container links-container">
        <div class="row">
          <div class="col-md-8 footer-links">
            <div class="row">
              <div class="col-md-3 col-sm-6">
                <h4 class="links-header">Departments</h4>
                <div class="col-md-12 links"><a href="#">Women</a> <a href="#">Men</a> <a href="#">Children</a> <a href="#">Home</a></div>
              </div>
              <div class="col-md-3 col-sm-6">
                <h4 class="links-header">User Profile</h4>
                <div class="col-md-12 links"><a href="#">Orders</a> <a href="#">Reviews</a> <a href="#">Coupons</a></div>
              </div>
              <div class="col-md-3 col-sm-6">
                <h4 class="links-header">Help</h4>
                <div class="col-md-12 links"><a href="#">How to order</a> <a href="#">How to pay</a> <a href="#">Delivery</a> <a href="#">Returns</a></div>
              </div>
              <div class="col-md-3 col-sm-6">
                <h4 class="links-header">About</h4>
                <div class="col-md-12 links"><a href="#">About Us</a> <a href="#">Privacy Policy</a> <a href="#">Career</a> <a href="#">Contact Us</a></div>
              </div>
            </div>
          </div>
          <div class="col-md-4 footer-social">
            <div class="text-right"><a href="#" class="social-icon"><i class="fa fa-facebook fa-2x"></i></a> <a href="#" class="social-icon"><i class="fa fa-instagram fa-2x"></i></a> <a href="#" class="social-icon"><i class="fa fa-pinterest-p fa-2x"></i></a> <a href="#" class="social-icon"><i class="fa fa-youtube-square fa-2x"></i></a></div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
